src/canisters/frontend/ic-asset/src/sync.rs

```diff
@@ -220,9 +220,17 @@ fn include_entry(entry: &walkdir::DirEntry, config: &AssetConfig) -> bool {
         .map(|s| s.starts_with('.'))
         .unwrap_or(false);
 
-    match (starts_with_a_dot, config.ignore) {
-        (dot, None) => !dot,
-        (_dot, Some(ignored)) => !ignored,
+    let well_known = entry.path().is_dir()
+        && entry
+            .file_name()
+            .to_str()
+            .map(|s| s.starts_with(".well-known"))
+            .unwrap_or(false);
+
+    match (starts_with_a_dot, config.ignore, well_known) {
+        (_, ignored, wk) if wk && !ignored.map_or(false, |v| v) => true,
+        (dot, None, _) => !dot,
+        (_dot, Some(ignored), _) => !ignored,
     }
```

Comment:
Given that the logic here is, in order:
1. If the `ignore` field is present, use that
2. If it's in a `.well-known` directory, include it
3. include files that do not begin with a dot

Three ideas to make this easier to understand:

1. Reorder the match parameters left-to-right, cases top-to-bottom, and streamline the expressions:
```suggestion
    match (config.ignore, well_known, starts_with_a_dot) {
        (Some(ignored), _, _) => !ignored,
        (None, true, _) => true,
        (None, _, starts_with_a_dot) => !starts_with_a_dot,
    }
```

2. In this case an if-else chain might be easiest to understand:
```suggestion
    if let Some(ignore) = config.ignore {
        !ignore
    } else if well_known {
        true
    } else {
        !starts_with_a_dot
    }
```

3. Finally, while this method returns true to `include_entry`, this piece might be easier to understand by inverting that logic here:
```suggestion
    let ignore = if let Some(ignore) = config.ignore {
        ignore
    } else if well_known {
        false
    } else {
        starts_with_a_dot
    };
    !ignore
```